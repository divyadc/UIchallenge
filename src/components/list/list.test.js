import ListComponent from "./index.js";
import React from 'react';
import { shallow } from 'enzyme';

describe('ListComponent', () => {
    it('Should render correctly in "debug" mode', () => {
        const component = shallow(<ListComponent debug />);
        expect(component).toMatchSnapshot();
    });

    it('ButtonClickEvent', () => {
        const mockCallBack = jest.fn();
        const button = shallow((<button onClick={mockCallBack}>Show More</button>));
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    // it('Check Search Data', () => {
    //     const listComponent = new ListComponent();
    //     let result = jest.fn(() => {
    //         return Promise.resolve(listComponent.getSearchData("sea", 1));
    //     });
    //     expect(result.mock.instances.length).toBeGreaterThanOrEqual(1);
    // })
});

