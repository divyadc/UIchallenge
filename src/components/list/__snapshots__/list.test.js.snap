// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent Should render correctly in "debug" mode 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search or jump to..."
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;

exports[`ListComponent should render correctly in "debug" mode 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search or jump to..."
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
